File: Python/Facade.py
# Підсистема 1
class Subsystem1:
    def operation1(self):
        print("Subsystem1: Operation1")


# Підсистема 2
class Subsystem2:
    def operation2(self):
        print("Subsystem2: Operation2")


# Підсистема 3
class Subsystem3:
    def operation3(self):
        print("Subsystem3: Operation3")


# Фасад
class Facade:
    def __init__(self):
        self.subsystem1 = Subsystem1()
        self.subsystem2 = Subsystem2()
        self.subsystem3 = Subsystem3()

    def operation_facade(self):
        print("Facade: OperationFacade")
        self.subsystem1.operation1()
        self.subsystem2.operation2()
        self.subsystem3.operation3()


# Клієнт
def main():
    facade = Facade()
    facade.operation_facade()


if __name__ == "__main__":
    main()

File: Facade/Facade/Facade.cs
using Facade.Subsystems;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Facade
{
    internal class Facade
    {
        private Subsystem1 subsystem1;
        private Subsystem2 subsystem2;
        private Subsystem3 subsystem3;

        public Facade()
        {
            subsystem1 = new Subsystem1();
            subsystem2 = new Subsystem2();
            subsystem3 = new Subsystem3();
        }

        public void OperationFacade()
        {
            Console.WriteLine("Facade: OperationFacade");
            subsystem1.Operation1();
            subsystem2.Operation2();
            subsystem3.Operation3();
        }
    }
}
